kmymoney: fix build for Qt 6.10

das-g opened the pull request:
Fix KMyMoney build for (I presume from the patch name) Qt 6.10 by adding patch `kmymoney-fix-build-against-qt-6-10.patch` from <https://src.fedoraproject.org/rpms/kmymoney/blob/rawhide/f/kmymoney-fix-build-against-qt-6-10.patch>. (See https://github.com/NixOS/nixpkgs/issues/455948#issuecomment-3448763451.)

Resolves #455948


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SigmaSquadron commented on pkgs/by-name/km/kmymoney/package.nix:
```diff
@@ -77,6 +77,10 @@ stdenv.mkDerivation (finalAttrs: {
 
+  patches = [
+    ./kmymoney-fix-build-against-qt-6-10.patch
+  ];
```
Since this is Fedora's patch, why are we not using `fetchpatch`?

SigmaSquadron replied:
Yes. We don't vendor patches unless necessary.

das-g replied:
Changed to `fetchpatch` with https://github.com/NixOS/nixpkgs/compare/4ea4f4a28cf453f11d50d04dcdde05518baa3664..e9560c0661aa8d635ced15d30cfda3b13bfc41ed

das-g commented:
> Thanks for addressing the build issue; the changes look good to me. While running nixpkgs-review, the build initially failed with a 503 error when fetching the patch. The URL appears to be protected by [Anubis](https://github.com/TecharoHQ/anubis). After I opened the patch URL in my browser and passed the Anubis verification, the build completed successfully (likely because my IP was then whitelisted).

Yeah, I've noticed the Anubis screen when first (then manually) accessing the website, too. As I wasn't aware Anubis whitelists IPs, I thought it was fine anyway when e9560c0661aa8d635ced15d30cfda3b13bfc41ed successfully built locally for me. I probably naïvely thought that might be due to different user-agent values, and then didn't quite realize yet that that would defeat Anubis' purpose, as crappy AI scrapers of course can't reliably be blocked by relying on user-agent strings.

> This could potentially cause build failures on Hydra as well, since it might not have access to the protected patch URL.

So, back to vendoring the patch, then, or are there any better options?

If back to vendoring, is 4ea4f4a28cf453f11d50d04dcdde05518baa3664 good, or should the patch source also be credited in-tree rather than just in the commit message?

das-g committed:
kmymoney: fix build for Qt 6.10

by adding patch
from https://src.fedoraproject.org/rpms/kmymoney/blob/rawhide/f/kmymoney-fix-build-against-qt-6-10.patch

Resolves #455948
